spec/node_spec.rb

```diff
@@ -65,20 +65,19 @@ module Ferrum
       end
     end
 
-    context "when the element is not in the viewport of parent element", skip: true do
+    context "when the element is not in the viewport of parent element" do
       before do
         browser.go_to("/ferrum/scroll")
       end
 
       it "scrolls into view" do
-        # FIXME:
-        browser.at_xpath("//a[text() = 'Link outside viewport']").click
```

Comment:
@route is the idea was: use `click` without options even if the element is outside of viewport?

like this:
```
if element_outside_of_viewport?
  evaluate("this.click()")
  return self
end

def element_outside_of_viewport?
  evaluate("this.getBoundingClientRect().bottom < 0") ||
    evaluate("this.getBoundingClientRect().right < 0") ||
    evaluate("this.getBoundingClientRect().left > (window.innerWidth || document.documentElement.clientWidth)") ||
    evaluate("this.getBoundingClientRect().top > (window.innerHeight || document.documentElement.clientHeight)")
end
```

based on https://stackoverflow.com/questions/123999/how-can-i-tell-if-a-dom-element-is-visible-in-the-current-viewport/7557433#7557433

Reply:
Check `cuprite` on how it's handled there. I think main difference is that `cuprite` scrolls into viewport automatically because it's a complicated driver, `ferrum` instead should be close to what real user would do with the browser. So if you can't see a node in the viewport you can't click on it, but we might be missing some API for convenient scrolling then.. I think we can drop these tests in general after having something for scrolling but I left them as a reminder to implement it.